specs/dsl.md

```diff
@@ -0,0 +1,150 @@
+# DSL 
+DSL is the *domain specific language* used to communicate with a single zero-os node. The language is designed to 
+allocate and build resources with great flexibility. The DSL language is like an instruction blueprint where user
+can use it to pull different type of primitives together to build a working solution.
+
+# Content-Type
+The API is probably going to support multiple content-types, json, yaml, etc... in the rest of this document we going to use `yaml` to
+show structures, and objects for readability.
+
+# Philosophy
+Here we gonna list what the DSL is all about and how it should behave. This is an open discussion area. This is basically the ideas on 
+how the system should interface with the world. Hence it's more of the philosophy of zos, rather than the specs. The following rules is
+going to define how the entire system should behave.
+
+- The main entity of the API is a `resource`, you can create, list, inspect and destroy a resource
```

Comment:
> you can create, list, inspect and destroy a resource

That's not right, the only thing you can do is create or destroy. There is not listing or inspection.
There will be some possibility to get monitoring information for the resources a client have reserved and own. But in no case a user will be able to do a listing of the resources present on a node.